eip-5564.md

```diff
@@ -118,6 +118,11 @@ contract IERC5564Messenger is StakeManager{
   /// Besides the view tag, the metadata can be used by the senders however they like, 
   /// but the below guidelines are recommended:
   /// The first byte of the metadata MUST be the view tag.
+  /// - When sending the native token of the blockchain (cf. ETH), the metadata SHOULD be structured as follows:
+  ///     - Byte 1 MUST be the view tag, as specified above.
+  ///     - Bytes 2-5 are arbitrary bytes (e.g. 0xdeadbeef).
+  ///     - Bytes 6-25 are the address 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE.
+  ///     - Bytes 26-57 are the amount of ETH being sent.
   /// - When sending ERC-20 tokens, the metadata SHOULD be structured as follows:
   ///   - Byte 1 MUST be the view tag, as specified above.
   ///   - Bytes 2-5 are the method Id, which the hash of the canonical representation of the function to call.
```

Comment:
We use the approve selector below, so we probably should have this also?
```suggestion
  ///   - Bytes 2-5 are 0xa9059cbb, which the signature for the ERC-20 transfer method
```

Reply:
You're right, but I wouldn't reduce it to transfers and approvals, but instead keep it open for any kind of interaction. So that even a mint to a stealth address can be signaled.

So, I'd prefer keeping the method id.

Reply:
Yo @mds1, I've tried to generalize it more such that we only have to differentiate between native tokens and tokens from standards (ERC-20, ERC-721, ect.).

Let me know what you think about it.

Reply:
maybe we update this bullet to this? pulled the definition from the solidity docs so it's unambiguous, and I say "often, but not always" because contracts in other languages don't necessarily have to follow that same convention
```suggestion
  ///   - Bytes 2-5 are the function selector. This is often, but not always,
  //      defined as the first (left, high-order in big-endian) four bytes of
  //      the Keccak-256 hash of the signature of the function.
```